4.analyze-components/scripts/nbconverted/3.within-z-svcca-analysis.py

```diff
@@ -0,0 +1,93 @@
+
+# coding: utf-8
+
+# # Determining Within Z Dimension Stability with SVCCA
+# 
+# **Gregory Way 2018**
+# 
+# Here, we apply Singular Vector Canonical Correlation Analysis ([Raghu et al. 2017](https://arxiv.org/abs/1706.05806 "SVCCA: Singular Vector Canonical Correlation Analysis for Deep Learning Dynamics and Interpretability")) ([github](https://github.com/google/svcca)) to the neuron matrices Z to quantify model stability both within and across algorithms over each bottleneck dimensionality.
+# 
+# Briefly, SVCCA uses Singular Value Decomposition (SVC) to extract the components explaining 99% of the variation.
+# This is done to remove potential dimensions described by noise.
+# Next, SVCCA performs a Canonical Correlation Analysis (CCA) on the SVD matrices to identify maximum correlations of linear combinations of both input matrices. The algorithm will identify the canonical correlations of highest magnitude across and within algorithms of the same dimensionality. 
+# 
+# The output of the SVCCA analysis is the SVCCA mean similarity score. This single number can be interpreted as a measure of similarity, or stability, in the solutions identified from each compression algorithm.
+
+# In[1]:
+
+
+import os
+import glob
+import numpy as np
+import pandas as pd
+
+from scripts.util import read_in_z, get_svcca_across_algorithm_stability
+
+
+# In[2]:
+
+
+datasets = ['TARGET', 'TCGA', 'GTEX']
+algorithms = ['pca', 'ica', 'nmf', 'dae', 'vae']
+z_dims = [2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 14, 16, 18, 20, 25, 30,
+          35, 40, 45, 50, 60, 70, 80, 90, 100, 125, 150, 200]
+signals = (True, False)
+
+
+# In[3]:
+
+
+large_svcca_results_list = []
+for dataset in datasets:
+
+    for z in z_dims:
+
+        for signal in signals:
+
+            if signal:
+                shuffled_status = 'signal'
+            else:
+                shuffled_status = 'shuffled'
+
+            print("Calculating... dataset {} for {} dimension {}"
+                  .format(dataset, shuffled_status, z))
+
+            # Read in the z matrix of interest
+            z_dict = read_in_z(
+                dataset=dataset,
+                z_dim=z,
+                algorithm='all',
+                shuffled_data=signal
+            )
+
+            # Perform across algorithm SVCCA
+            svcca_out = (
+                get_svcca_across_algorithm_stability(z_dict=z_dict,
```

Comment:
so this function isn't included in this script? did nbconvert miss it?

Reply:
its imported from [`scripts/util.py`](https://github.com/greenelab/interpret-compression/pull/27/files#diff-8e182ee61a3884cb05d84bddb239ab3bR57)